dapps/marketplace/src/pages/listing/_BuyFractional.js

```diff
@@ -11,125 +11,185 @@ import PaymentOptions from './_PaymentOptions'
 
 import Buy from './mutations/Buy'
 import DateRange from './_DateRange'
+import ConfirmShippingAndPurchase from './_ConfirmShippingAndPurchase'
+import PurchaseSummary from './_PurchaseSummary'
 
+const withFractionalData = WrappedComponent => {
+  const WithFractionalData = ({ listing, range, availability, ...props }) => {
+    const acceptsDai = listing.acceptedTokens.find(t => t.id === 'token-DAI')
+    const token = acceptsDai ? 'token-DAI' : 'token-ETH'
+
+    let startDate = null,
+      endDate = null,
+      totalPrice,
+      available = false
+
+    if (range) {
+      const split = range.split('/')
+      startDate = split[0]
+      endDate = split[1]
+      const priceEstimate = availability.estimateNightlyPrice(range)
+      available = priceEstimate.available
+      if (available) {
+        totalPrice = {
+          amount: String(priceEstimate.price),
+          currency: listing.price.currency
+        }
+      }
+    }
+
+    return (
+      <WithPrices
+        price={totalPrice}
+        target={token}
+        targets={['token-ETH', 'token-DAI', listing.price.currency.id]}
+        allowanceTarget={listing.contractAddr}
+      >
+        {({ prices, tokenStatus }) => (
+          <WrappedComponent
+            {...{
```

Comment:
Don't think we need these extra ...?

Reply:
We need that. I should have done something like `<WrappedComponent {...props} prices={prices} tokenStatus={tokenStatus}  />` though. Will change it.